WinRT.Runtime/Projections/Type.cs

```diff
@@ -85,8 +91,13 @@ public static Type GetAbi(Marshaler m)
             *(Type*)dest.ToPointer() = GetAbi(arg);
 
         public static Type FromManaged(global::System.Type value)
-        {
-            return GetAbi(CreateMarshaler(value));
+        {
+            var abi = ToAbi(value);
+            return new Type
+            {
+                Name = MarshalString.FromManaged(abi.Name),
```

Comment:
The effect of this change I see is that we are now doing WindowsCreateString rather than WindowsCreateStringReference in this scenario.  Do we know why WindowsCreateStringReference doesn't work here, but works above?  And shouldn't MarshalString.GetAbi work for both scenarios assuming it isn't WindowsCreateStringReference just failing?

Reply:
WindowsCreateStringReference is wrong and was causing crashes, because there's nothing to hold the marshaler for the lifetime of the returned Type object.  In other words, the entire pattern is wrong.  We were using the 'in' pattern of creating a marshaler and then passing its value across the ABI.  This is an 'out' pattern, where we must allocate the entire object for the ABI.

Reply:
thanks, that makes sense.